guides/plugins/plugins/using-custom-fields.md

```diff
@@ -0,0 +1,103 @@
+# Using custom fields
+
+## Overview
+
+Shopware's custom field system allows you to extend entities, without writing a complete entity extension.
+This is possible by storing the additional data in a [JSON-Field](https://dev.mysql.com/doc/refman/8.0/en/json.html).
+If you want to learn more about custom fields or even add your own custom fields to entities you might want to take a look at this guide: [Add custom input field to existing component](administration\add-custom-field.md).
+
+This guide will cover how to write to custom fields with or with out the `sw-custom-field-set-renderer`
+
+## Prerequisites
+
+All you need for this guide is a running Shopware 6 instance, the files and preferably a registered module. Of course you'll have to understand JavaScript and have a basic familiarity with VueJS. However, that's a prerequisite for Shopware as a whole and will not be taught as part of this documentation.
+
+## Preparing the template
+
+First we are creating a template to use the `sw-custom-field-set-renderer`, you don't have to do this but, it makes it easier to see the `custom fields`
+
+```html 
+<div>
+    <sw-custom-field-set-renderer
+        :entity="product"
+        showCustomFieldSetSelection
+        :sets="sets">
+    </sw-custom-field-set-renderer>
+    <button @click="save">Save</button>
+</div>
+```
+
+## Loading the custom fields
+
+The following code sample shows how to load and save the the custom field. This code sample is tailored to the `sw-custom-field-set-renderer`, but it's not essential to use the `sw-custom-field-set-renderer`. The custom fields are loaded automatically and it's possible to just modify the `custom-fields` without loading the schema, because the schema is just needed for display purposes.
+
+```javascript
```

Comment:
This example is quite big. Please use comments to emphasize the important lines. Or if are the two lines the only important ones, could we shorten the example a little bit? Only if possible :)

Reply:
i've added a few comments  i hope it's enough